### PR TITLE
Added all types of expect

### PR DESCRIPTION
**Pull request by UnDarkle**

### CODE REVIEW

File: tests/core/utils.py [hunk 1]
```diff
@@ -321,8 +322,24 @@ class StoreFailureCase(MockCase):
...
     def test_store_failure(self):
         """Mock test function - stores failures."""
-        self.expect(False, self.FAILURE_MESSAGE)
-        self.assertTrue(self.ASSERTION_MESSAGE)
+        self.expectTrue(False, self.FAILURE_MESSAGE)
+        self.assertTrue(False, self.ASSERTION_MESSAGE)
+
+
+class ExpectRaisesCase(MockCase):
+    """Mock case, expect multiple exceptions."""
```
Comment (rinslow) [inline tests/core/utils.py]: Insert more documentation, maybe something along these lines
```python
"""Mock case that accumulates errors using expectRaises both as a function call and as a context-manager. """
```

File: tests/core/test_case.py [hunk 2]
```diff
@@ -169,14 +176,14 @@ def test_success_case_run(self):
...
 
-        test_resource = case.all_resources[self.DEMO_RESOURCE_NAME]
-        self.assertTrue(isinstance(test_resource, DemoResource),
-                "State resource data type should have been 'DemoResourceData'")
-
         self.assertEqual(case.data.exception_type, TestOutcome.SUCCESS,
                          "Unexpected test outcome, expected %r got %r" %
                          (TestOutcome.SUCCESS, case.data.exception_type))
 
+        test_resource = case.all_resources[self.DEMO_RESOURCE_NAME]
```
Comment (rinslow) [inline tests/core/test_case.py]: why did you move this block?

File: tests/core/test_case.py [hunk 3]
```diff
@@ -357,6 +366,66 @@ def test_expect_and_assert_case_run(self):
...
+    def test_expect_raises_and_assert_case_run(self):
+        """Test a TestCase that uses both expect raises and assert.
+
+        * Defines the registered resource as required resource.
+        * Runs the test under a test suite.
+        * Validates that the test failed with three tracebacks.
+        * Validates the case's data object.
+        * Validates the resource's state.
+        """
+        TempExpectRaisesCase.resources = (request('test_resource',
```
Comment (rinslow) [inline tests/core/test_case.py]: isnt this redundant? you already requested resources in the TempExpectRaisesCase

File: src/rotest/core/abstract_test.py [hunk 4]
```diff
@@ -331,3 +330,146 @@ def store_state(self):
 
         for resource in self.all_resources.itervalues():
             resource.store_state(store_dir)
+
+    def _wrap_assert(self, assert_method, *args, **kwargs):
+        try:
+            assert_method(*args, **kwargs)
+
+        except AssertionError as err:
+            self.expect(False, str(err))
```
Comment (rinslow) [inline src/rotest/core/abstract_test.py]: I don't like this part, 
the `self.expect(False)` bit to be exact, would the user get a weird Error such as `False is not True`?

Comment (rinslow) [reply src/rotest/core/abstract_test.py]: Also what about the traceback?

File: tests/core/test_case.py [hunk 5]
```diff
@@ -13,7 +13,7 @@
 from tests.core.utils import (ErrorInSetupCase, SuccessCase, FailureCase,
```
Comment (rinslow) [inline tests/core/test_case.py]: I would add unit-tests to all of the new functions you added. 
It's probably painstaking work but it will A. affect our coverage, B. save us from future bugs

File: src/rotest/core/abstract_test.py [hunk 6]
```diff
@@ -104,8 +105,7 @@ def release_resource_loggers(self):
     @classmethod
     def get_resource_requests_fields(cls):
         """Yield tuples of all the resource request fields of this test.
-
-        Yields:
+         Yields:
```
Comment (osherdp) [inline src/rotest/core/abstract_test.py]: no

File: tests/core/test_case.py [hunk 7]
```diff
@@ -357,6 +366,66 @@ def test_expect_and_assert_case_run(self):
...
+        match = re.search(expected_traceback, case.data.traceback,
+                          flags=re.DOTALL)
+
+        self.assertIsNotNone(match,
+                     "Unexpected traceback, %r doesn't match the expression %r"
+                     % (case.data.traceback, expected_traceback))
+
+        test_resource = case.all_resources[self.DEMO_RESOURCE_NAME]
+
+        self.assertTrue(isinstance(test_resource, DemoResource),
```
Comment (osherdp) [inline tests/core/test_case.py]: here and everywhere: indentation problems

File: src/rotest/core/abstract_test.py [hunk 8]
```diff
@@ -331,3 +330,146 @@ def store_state(self):
...
+        except AssertionError as err:
+            self.expect(False, str(err))
+
+    def expectFalse(self, expr, msg=None):
+        self._wrap_assert(self.assertFalse, expr, msg)
+
+    def expectTrue(self, expr, msg=None):
+        self._wrap_assert(self.assertTrue, expr, msg)
+
+    def expectEqual(self, first, second, msg=None):
```
Comment (osherdp) [inline src/rotest/core/abstract_test.py]: add to readthedocs a usage of ```expectEqual```, so others know its existence
